### ISSUE TITLE
String double quotes in C get misinterpreted by pdflatex

### ISSUE THREAD

**Issue by the78mole** [doxygen/doxygen#7804]
**Describe the bug**

This is a code part that needs to get doxyfied:
```
enum2str_t cmd2str_lut[] = {
	{ C_NOP,   DIR_BIDIR, "NOP",  },
	{ C_CMD,   DIR_MOSI,  "CMD"   },
	{ C_STAT,  DIR_MISO,  "STAT"  },
	{ C_OK,    DIR_BIDIR, "OK"    },
	{ C_FAIL,  DIR_MISO,  "FAIL"  },
	{ C_BUSY,  DIR_MISO,  "BUSY"  },
// [...]
```

This is, what pdflatex creates from generated latex sources:
![image](https://user-images.githubusercontent.com/7723396/83013403-3f9d7180-a01d-11ea-8f52-bec785c0e574.png)

in HTML it looks as expected:
![image](https://user-images.githubusercontent.com/7723396/83013540-7a070e80-a01d-11ea-9d0e-7d4c7f236e74.png)

The latex code for the C_OK part looks like this:
```
\DoxyCodeLine{    \{ \mbox{\hyperlink{serial_8h_a00d51b3cb2e1fa09bdba2968e0cf7ac7a5dd35438b334486b45be8ef4c2e54a89}{C\_OK}},    \mbox{\hyperlink{serial_8h_a0d4ba43abfa4e330b05b0efd7f2d02e9a03c173d03473f59ec123fae188717d1d}{DIR\_BIDIR}}, \textcolor{stringliteral}{"OK"}    \},}
```

**Expected behavior**
I would expect, that PDF output looks mostly the same like HTML

**Screenshots**
Here is another example, where the closing quotes are somehow interpreted in a wrong way:
![image](https://user-images.githubusercontent.com/7723396/83013871-fef22800-a01d-11ea-8d30-c42aa5bdee09.png)

**To Reproduce**
Use a constant string somewhere, enclosed by double quotes. Latex will obviously try t

**Version**
1.8.18 (a1b07ad0e92e4526c9ba1711d39f06b58c2a7459)

**Stack trace**
no stack trace

**Additional context**
no further context

**Comment by albert-github** [comment]
You are using as output language "German" and when I use it I get a error during the PDF generation:
```
("D:\Program Files\MiKTeX 2.9\tex/latex/psnfss\t1pcr.fd")
! Argument of \language@active@arg" has an extra }.
<inserted text>
                \par
l.20 ...R, \textcolor{stringliteral}{"NOP"},  \},}

?
```

- Did you get this error as well? or did you use some extra settings like `LATEX_BATCHMODE`
- It is very beneficial to have a complete case and the settings used, so please add attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7805

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by the78mole** [comment]
Yes, I do use German settings. Your patch looks fine. Unfortunately, my build on mingw using `cmake -G  "Unix Makefiles" ..` fails at the link step of the final executable. So I cannot test your patch :-(

**Comment by albert-github** [comment]
@the78mole 
- did you already clear the problem regarding the linking?
- did you ask at e.g. https://stackexchange.com/ ?
- any messages from the `CMake` / `make` command?

Regarding the link issue please don't post the additional information here but either on https://stackexchange.com/  or in a new issue.

**Comment by the78mole** [comment]
Created QA on Stack OVerflow: https://stackoverflow.com/questions/62086016/building-doxygen-on-msys2

**Comment by the78mole** [comment]
Now I can confirm, that it's solved. Thanks!!!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.19.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
